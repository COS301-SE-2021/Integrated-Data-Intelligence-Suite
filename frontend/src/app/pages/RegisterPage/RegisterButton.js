import {Button} from 'antd';
import React, {Component} from 'react';
// import {validateLoginDetails} from "./functions/validateLoginDetails";

class RegisterButton extends React.Component {
    constructor(props) {
        super(props);
    }

    state = {
        is_loading: false,
        username: '',
        password: '',
    }

    enterLoading(val) {

        this.setState({
            is_loading: !this.state.is_loading,
        })

        this.setState({
            is_loading: false,
        })

        // console.log(user)
    }

    render() {
        return (
            <>
                <Button
                    type="primary"
                    htmlType="submit"
                    className="login_button"
                    loading={this.state.is_loading}
                    onClick={() => this.enterLoading(!this.state.is_loading)}
                >
                    Register
                </Button>
            </>
        );
    }
}

export default RegisterButton;