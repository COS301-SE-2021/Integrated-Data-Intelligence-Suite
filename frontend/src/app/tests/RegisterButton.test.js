import React from 'react';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import RegisterButton from '../components/RegisterButton/RegisterButton';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  const tree = renderer
    .create(<RegisterButton />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
