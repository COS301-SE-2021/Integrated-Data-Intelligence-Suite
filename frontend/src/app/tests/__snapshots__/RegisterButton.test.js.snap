// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="ant-btn ant-btn-primary login_button"
  onClick={[Function]}
  type="submit"
>
  <span>
    Register
  </span>
</button>
`;
